llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -292,6 +292,11 @@ class WaitcntBrackets {
     VgprVmemTypes[GprNo] = 0;
   }
 
+  void setNonKernelFunctionInitialState() {
+    setScoreUB(VS_CNT, getWaitCountMax(VS_CNT));
+    PendingEvents |= WaitEventMaskForInst[VS_CNT];
+  }
```

Comment:
Could you please split out a prerequisite patch that just adds this functionality? I think there are only a couple of tests that it will affect, to do with barriers or release vgprs.

Reply:
#75436